src/test/java/core/basesyntax/service/impl/ActivitiesProcessorImplTest.java

```diff
@@ -0,0 +1,128 @@
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import core.basesyntax.db.FruitsDb;
+import core.basesyntax.model.FruitActivity;
+import core.basesyntax.service.ActivitiesProcessor;
+import core.basesyntax.service.activity.strategy.ActivityHandler;
+import core.basesyntax.service.activity.strategy.BalanceHandler;
+import core.basesyntax.service.activity.strategy.PurchaseHandler;
+import core.basesyntax.service.activity.strategy.ReturnHandler;
+import core.basesyntax.service.activity.strategy.SupplyHandler;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class ActivitiesProcessorImplTest {
+
+    private static final Map<FruitActivity.Type, ActivityHandler> ACTIVITY_HANDLER_MAP =
+            new HashMap<>();
+
+    static {
+        ACTIVITY_HANDLER_MAP.put(FruitActivity.Type.BALANCE, new BalanceHandler());
+        ACTIVITY_HANDLER_MAP.put(FruitActivity.Type.SUPPLY, new SupplyHandler());
+        ACTIVITY_HANDLER_MAP.put(FruitActivity.Type.PURCHASE, new PurchaseHandler());
+        ACTIVITY_HANDLER_MAP.put(FruitActivity.Type.RETURN, new ReturnHandler());
+    }
+
+    private static final ActivitiesProcessor activitiesProcessor =
+            new ActivitiesProcessorImpl(new ActivityStrategyImpl(ACTIVITY_HANDLER_MAP));
+    private static final Map<String, Integer> fruitDb = FruitsDb.fruitDb;
+
+    @BeforeEach
+    void setUp() {
+        fruitDb.clear();
+    }
+
+    @ParameterizedTest
+    @MethodSource("validActivities")
+    void parseData_validActivities_Ok(
+            List<FruitActivity> activities, Map<String, Integer> expected) {
+        assertDoesNotThrow(() ->
+                        activitiesProcessor.processActivities(activities),
+                "Parsing valid lines shouldnt throw any exceptions");
+        Map<String, Integer> actual = fruitDb;
+        assertEquals(expected, actual, "FruitDB isn't equals expected");
+    }
+
+    static Stream<Arguments> validActivities() {
+        return Stream.of(
+                arguments(
+                        new ArrayList<FruitActivity>() {{
+                            add(new FruitActivity(FruitActivity.Type.BALANCE, "banana", 20));
+                            add(new FruitActivity(FruitActivity.Type.BALANCE, "apple", 100));
+                            add(new FruitActivity(FruitActivity.Type.SUPPLY, "banana", 100));
+                            add(new FruitActivity(FruitActivity.Type.PURCHASE, "banana", 13));
+                        }},
+                        new HashMap<String, Integer>() {{
+                            put("banana", 107);
+                            put("apple", 100);
+                        }}
```

Comment:
Here and then, the placement of the code should be corrected, in my opinion, there are too many code transfers to a new line. For example, you could not move arguments to a new line, it would look better

Reply:
Aggree. 
Also, use `Map.of` instead of constructor and put methods 
Fix it everywhere